### ISSUE TITLE
Configurable concurrent number in the same node for fs-backup and data mover

### ISSUE THREAD

**Issue by Lyndon-Li** [vmware-tanzu/velero#6663]
At present, the concurrent number for VGDP (Velero General Data Path) is hard-coded as 1, which affects the users of VGDP -- fs-backup and data mover. It is valuable to make this configurable by users. This involves:
1. Test and verify that VGDP is working well for concurrent number larger than 1
2. Create a global option for users to configure this number

**Comment by reasonerjt** [comment]
A design will be needed if we wanna set node-level limitation for concurrency.

**Comment by Lyndon-Li** [comment]
The concurrent number should be set on VGDP level, which applies to the pod volume backups and data movers in the same node.

This design will be based on the ALL-IN-ONE policy, we will add below snippet in the policy:
```
  vgcp:
    defaultConcurrentNum: 2
    concurrentNumByNode
      node-xxx: 3
      node-yyy: 8
```

**Comment by sseago** [comment]
@Lyndon-Li so to confirm, the above policy would set 3 for node-xxx, 8 for node-yyy, and 2 for everything else? So this would mean that in cases where we don't need node-specific differences, users would just set `defaultConcurrentNum` and that would apply to all nodes? But if there were certain nodes which were different in size or expected load than the others, they could tweak those particular node's concurrency numbers to reflect that?

**Comment by Lyndon-Li** [comment]
> @Lyndon-Li so to confirm, the above policy would set 3 for node-xxx, 8 for node-yyy, and 2 for everything else? So this would mean that in cases where we don't need node-specific differences, users would just set `defaultConcurrentNum` and that would apply to all nodes? But if there were certain nodes which were different in size or expected load than the others, they could tweak those particular node's concurrency numbers to reflect that?

Yes, this is the preliminary idea. This also means that the configuration will be more complex than what could fit into the node-agent server CLI parameter, so we will need a separate configuration, something like a global node-agent configuration, if the ALL-IN-ONE policy is not going to delivered.

**Comment by reasonerjt** [comment]
The configuration may need more discussion but let's make sure we have a design in v1.13.0

**Comment by shawn-hurley** [comment]
1. if we have node-specific limits, can we please use labels/label selectors for this instead of the node name? I am pretty sure that the node name is also a label, so you could still choose one node based on the label.
2. Can we explain why we need the ability to do this per node?

**Comment by sseago** [comment]
@shawn-hurley "Can we explain why we need the ability to do this per node?"

I'm not 100% sure whether there is much in the way of real-world use cases for this -- certainly in most cases we would not need this param and just want to set concurrency to N on each node. But I can imagine certain edge cases -- maybe certain nodes are significantly larger and/or less loaded than the rest and can safely handle a greater concurrency, or the reverse -- certain nodes with smaller capacity which need a lower number.

**Comment by Lyndon-Li** [comment]
There are two considerations to differentiate the concurrent number across nodes:
1. The node capacity is not the same, as what Scott has mentioned
2. Computing resource/NIC are not shared across nodes --- workloads can only use the computing resource/NIC from where they are running. E.g., a workload may be more critical than others but requires large network bandwidth, in this case, users may not want data movers to run in the same node

Anyway, let's discuss this, if most of us think the aforementioned reasons for differentiation are advanced considerations, we can have the default concurrent number first in 1.13 and whenever necessary, we come back to add the advanced options.

**Comment by Lyndon-Li** [comment]
> labels/label selectors for this instead of the node name

Yes, will do, this is a good idea.

**Comment by shawn-hurley** [comment]
Man, I am sorry for missing both responses. I think that both cases make sense and if it is not too much trouble to add these extra knobs I think it makes a ton of sense! Thanks for explaining the use cases.

**Comment by Lyndon-Li** [comment]
@sseago @shawn-hurley Design PR has been submitted, please help to review it.

**Comment by Lyndon-Li** [comment]
Hold on closing, doc is required

**Comment by Lyndon-Li** [comment]
Fixed by #7059, #7161
